hole/floyd-steinberg-dithering.go

```diff
@@ -0,0 +1,12 @@
5\n154\n140\n6\n4\n161\n207\n214\n178\n8\n226\n232\n227\n216\n182\n160\n177\n196\n83\n35\n41\n73\n54\n55\n65\n66\n73\n80\n87\n89\n97\n106\n111\n119\n122\n127\n130\n134\n135\n137\n138\n137\n137\n135\n134\n132\n131\n125\n120\n114\n108\n100\n95\n90\n85\n80\n78\n72\n67\n110\n113\n119\n120\n129\n142\n135\n2\n4\n138\n141\n147\n186\n187\n9\n106\n82\n26\n18\n18\n18\n21\n27\n30\n83\n83\n60\n48\n52\n60\n65\n69\n73\n79\n84\n88\n96\n103\n107\n111\n114\n119\n122\n124\n129\n129\n129\n128\n124\n123\n120\n117\n112\n109\n104\n97\n92\n86\n85\n80\n74\n72\n65\n60\n92\n106\n122\n137\n201\n232\n4\n9\n213\n202\n176\n178\n161\n144\n45\n8\n11\n13\n17\n18\n17\n18\n23\n28\n50\n40\n39\n45\n47\n54\n59\n66\n70\n72\n79\n82\n85\n89\n94\n100\n104\n107\n109\n112\n113\n113\n113\n113\n113\n112\n109\n106\n103\n97\n92\n88\n84\n80\n78\n72\n66\n63\n61\n58\n86\n127\n195\n198\n163\n33\n0\n6\n3\n3\n3\n6\n6\n7\n7\n9\n9\n11\n14\n17\n19\n18\n23\n26\n32\n34\n38\n40\n48\n51\n54\n59\n64\n68\n73\n73\n78\n82\n85\n85\n92\n96\n98\n101\n103\n100\n103\n103\n103\n100\n97\n92\n89\n88\n84\n79\n77\n73\n72\n66\n61\n60\n56\n54\n192\n166\n27\n0\n0\n0\n0\n0\n0\n0\n2\n5\n5\n6\n8\n9\n9\n12\n14\n15\n19\n20\n21\n26\n28\n34\n35\n41\n44\n48\n49\n53\n57\n61\n66\n67\n73\n73\n77\n79\n82\n85\n86\n88\n88\n88\n89\n89\n89\n86\n86\n84\n82\n79\n78\n73\n72\n66\n63\n60\n58\n55\n52\n50\n119\n30\n0\n0\n0\n0\n0\n0\n0\n0\n2\n3\n3\n5\n6\n9\n10\n10\n14\n16\n17\n21\n22\n23\n27\n30\n34\n37\n41\n43\n48\n49\n54\n55\n61\n64\n66\n66\n72\n74\n73\n78\n78\n79\n80\n79\n79\n79\n80\n79\n79\n74\n72\n73\n72\n66\n63\n60\n60\n56\n54\n50\n48\n42\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n2\n3\n3\n5\n7\n10\n12\n12\n16\n18\n18\n24\n24\n25\n28\n31\n34\n39\n42\n41\n44\n50\n49\n53\n55\n57\n60\n63\n65\n67\n67\n69\n69\n72\n74\n74\n74\n73\n72\n69\n67\n65\n66\n64\n59\n55\n53\n51\n55\n48\n43\n43\n37\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n4\n4\n6\n8\n10\n12\n12\n17\n19\n18\n24\n25\n29\n28\n32\n35\n38\n37\n42\n43\n48\n48\n48\n54\n53\n58\n60\n60\n61\n64\n63\n66\n65\n65\n63\n63\n63\n61\n60\n59\n57\n54\n53\n51\n48\n47\n44\n42\n42\n39\n34\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n4\n5\n7\n6\n8\n11\n13\n12\n17\n19\n19\n24\n25\n25\n29\n28\n31\n34\n37\n42\n41\n43\n48\n49\n49\n51\n51\n53\n54\n55\n55\n54\n54\n54\n54\n54\n55\n54\n53\n53\n51\n49\n52\n49\n44\n42\n42\n37\n37\n35\n31\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n3\n4\n5\n7\n9\n11\n10\n11\n12\n15\n19\n18\n25\n23\n24\n25\n28\n31\n34\n36\n37\n38\n41\n42\n44\n43\n47\n49\n48\n47\n48\n48\n48\n48\n48\n48\n48\n47\n48\n48\n47\n44\n43\n42\n37\n37\n36\n34\n32\n29\n24\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n4\n4\n4\n7\n9\n8\n12\n12\n16\n19\n18\n18\n23\n23\n23\n25\n24\n27\n31\n34\n36\n37\n36\n41\n37\n40\n43\n42\n43\n49\n47\n47\n47\n47\n48\n43\n43\n42\n41\n40\n41\n37\n37\n33\n34\n31\n27\n24\n23\n23\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n4\n5\n7\n6\n8\n10\n11\n11\n12\n16\n16\n19\n22\n23\n23\n23\n24\n27\n27\n30\n31\n33\n33\n35\n36\n36\n36\n37\n40\n36\n36\n36\n36\n40\n40\n35\n36\n36\n36\n35\n33\n31\n32\n27\n27\n23\n23\n23\n23"},
```

Comment:
Rather than these giant Go strings I think it would be better to use the `hole/fixed-tests` functionality, see Emojify or others for details. It's simply a text file or alternating input/outputs (assuming one to one and assuming no double newlines). Then the Go just becomes:
```go
return outputMultirunTests(fixedTests("floyd-steinberg-dithering"))
```

Reply:
Or
```go
outputTests(shuffle(fixedTests("floyd-steinberg-dithering")))
```
if you want to run them all at once actually.

Reply:
I actually want to run each of them in its own separate run. Apparently [J can't handle more than 30 kb of command line arguments](https://discord.com/channels/756829356155731988/758357767177240596/1175871741181055066), which would be a problem if I tried to put all the images in one run. The maximum size of an input is about 16 kb, but in practice they are a bit smaller due to having some pixel values <100.

Alternatively, more images could fit in one run if they are smaller (perhaps 32x32 instead of 64x64). Or maybe just one 64x64 image would be enough?

Reply:
Ah in that case the Go would be something like this (untested):
```go
tests := fixedTests("floyd-steinberg-dithering")
runs := make([]Run, len(tests))
for i, test := range tests {
	runs[i] = Run{Args: []string{test.in}, Answer: test.out}
}
return shuffle(runs)
```
The point of using "fixed tests" is to keep the source code readable by moving the fixed test cases out into a text file. It's orthogonal to how exactly we structure the test cases into runs.